### DISCUSSION TITLE
Add a generic to joyride component to be able to type step data

### DISCUSSION THREAD

**Discussion by darkouz** [gilbarbara/react-joyride#1055] [Ideas]
Hi, for a personal project I patched the library to suit my needs :
The type of data in the step object is type as any today.
What would you think to allow to use a generic for the joyride component to be able to type the data step, like that it's more type safe, anywhere with use this key, like in the callback or in the tooltip.

`<ReactJoyride<StepData>`
`(data: CallBackProps<StepData>) => {`
<img width="463" alt="image" src="https://github.com/user-attachments/assets/b0f6fd65-70f1-4f11-abfb-23b476ac5cf1">

**Comment by gilbarbara** [comment]
Not planned. It's easy to cast the data locally.
